### PR TITLE
Postman Sync - caching + fast sync

### PR DESCRIPTION
**Pull request by thim81**
- [x] Improved caching of Postman API, reducing API calls towards Postman
- [x] Added fast Postman sync option to sync collections faster by using a delete & create of a new collection (new UID), instead of an update
- [x] Added Postman sync option to refresh all local cached Postman API data

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -75,6 +83,11 @@ require('dotenv').config()
       describe: 'Upload generated collection to Postman, after conversion',
       type: 'boolean'
     })
+    .option('syncPostman', {
+      alias: 'syncPostman',
```
Comment (nicklloyd) [inline src/index.ts:87]: feels like this could just be moved up above postmanFastSync

File: src/services/PostmanApiService.ts [hunk 2]
```diff
@@ -278,6 +278,83 @@ export class PostmanApiService {
     }
   }
 
+  async deleteCollection(postmanUid: string, workspaceId?: string): Promise<string> {
+    const workspaceIdParam = workspaceId ? `?workspace=${workspaceId}` : ''
```
Comment (nicklloyd) [inline src/services/PostmanApiService.ts:282]: think we can remove the worspaceId

File: src/services/PostmanApiService.ts [hunk 3]
```diff
@@ -278,6 +278,83 @@ export class PostmanApiService {
...
+        response = error?.response
+      }
+
+      const respData = response?.data ?? {}
+
+      if (responseStatusCode < 300) {
+        spinner.succeed(`Delete from Postman Succeeded`)
+      } else {
+        spinner.succeed(
+          `Delete from Postman completed with status: ${responseStatusCode}. \n\n Please review your collection within Postman as they can respond with a 5xx but still delete your collection`
```
Comment (nicklloyd) [inline src/services/PostmanApiService.ts:340]: does this still happen? ie: 5xx response

File: src/services/PostmanRepo.ts [hunk 4]
```diff
@@ -20,24 +22,40 @@ export class PostmanRepo {
...
         collections: [],
         workspaces: []
       }
       await this.persistCache()
     }
 
-    await this.getCollections()
-    await this.getWorkspaces()
+    // Check cache expiration
+    const oneHour = 60 * 60 * 1000
```
Comment (nicklloyd) [inline src/services/PostmanRepo.ts:42]: nice one

File: src/services/PostmanSyncService.ts [hunk 5]
```diff
@@ -45,6 +51,15 @@ export class PostmanSyncService {
 
     this.portmanCollection = portmanCollection
     this.collectionName = collectionName || (portmanCollection?.info?.name as string)
+
+    this.postmanFastSync = postmanFastSync || false
```
Comment (nicklloyd) [inline src/services/PostmanSyncService.ts:55]: should use `??` instead of `||` when checking for bools that might be undefined

**Review by nicklloyd — APPROVED** [review]

File: src/services/PostmanApiService.ts (hunk 2 above)
Comment (thim81) [reply src/services/PostmanApiService.ts]: removed

File: src/index.ts (hunk 1 above)
Comment (thim81) [reply src/index.ts]: re-organised the options

File: src/services/PostmanApiService.ts (hunk 3 above)
Comment (thim81) [reply src/services/PostmanApiService.ts:339]: Don't think it does, I'm not sure how the API will respond.

File: src/services/PostmanSyncService.ts (hunk 5 above)
Comment (thim81) [reply src/services/PostmanSyncService.ts:55]: done

**Pull request merged**